datafusion/core/src/physical_optimizer/enforce_sorting.rs

```diff
@@ -392,7 +392,11 @@ fn analyze_immediate_sort_removal(
         let sort_input = sort_exec.input();
         // If this sort is unnecessary, we should remove it:
         if sort_input.equivalence_properties().ordering_satisfy(
-            sort_exec.properties().output_ordering().unwrap_or_default(),
+            &sort_exec
+                .properties()
+                .output_ordering()
+                .cloned()
+                .unwrap_or_default(),
```

Comment:
I think if we had `LexOrdering::empty()` as explained below, we could avoid this clone like this
```rust
            &sort_exec
                .properties()
                .output_ordering()
                .unwrap_or(LexOrdering::empty()),
```

There are similar things below too